app/assets/javascripts/repositories/validators/base_validator.js

```diff
@@ -33,3 +33,124 @@ $.fn.dataTable.render.RepositoryAssetValueValidator = function($input) {
+  let $dateS = $container.find('.start-time input[data-datetime-part=date]');
+  let $timeS = $container.find('.start-time input[data-datetime-part=time]');
+  let $dateE = $container.find('.end-time input[data-datetime-part=date]');
+  let $timeE = $container.find('.end-time input[data-datetime-part=time]');
+  let isValid = true;
+  let errorMessage;
+  let startTime;
+  let endTime;
+  let a = [];
+
+  if ($input.val()) {
+    startTime = new Date(JSON.parse($input.val()).start_time);
+    endTime = new Date(JSON.parse($input.val()).end_time);
+  }
+
+  a.push($dateS.val() === '');
+  a.push($timeS.val() === '');
+  a.push($dateE.val() === '');
+  a.push($timeE.val() === '');
+
+  if (a.filter((v, i, arr) => arr.indexOf(v) === i).length > 1) {
+    isValid = false;
+    errorMessage = 'Needs to set all or none';
+  } else if (($input.val()) && (startTime > endTime)) {
+    isValid = false;
+    errorMessage = 'End time is before start time';
+  }
+
+  if (isValid) {
+    return true;
+  }
+
+  $container.addClass('has-error');
+  $container.append(`<span class="help-block">${errorMessage}</span>`);
+  return false;
+};
+
+$.fn.dataTable.render.RepositoryDateRangeValueValidator = function($input) {
+  // will be refactored
+
+  let $container = $input.parents().eq(1);
+  let $dateS = $container.find('.start-time input[data-datetime-part=date]');
+  let $dateE = $container.find('.end-time input[data-datetime-part=date]');
+  let isValid = true;
+  let errorMessage;
+  let endTime;
+  let startTime;
+  if ($input.val()) {
+    startTime = new Date(JSON.parse($input.val()).start_time);
+    endTime = new Date(JSON.parse($input.val()).end_time);
+  }
+
+  if (!($dateS.val() === '') === ($dateE.val() === '')) {
+    isValid = false;
+    errorMessage = 'Needs to set all or none';
+  } else if (($input.val()) && (startTime > endTime)) {
+    isValid = false;
+    errorMessage = 'End date is before start date';
+  }
+
+  if (isValid) {
+    return true;
+  }
+
+  $container.addClass('has-error');
+  $container.append(`<span class="help-block">${errorMessage}</span>`);
+  return false;
+};
+
+$.fn.dataTable.render.RepositoryTimeRangeValueValidator = function($input) {
+  // will be refactored
+
+  let $container = $input.parents().eq(1);
+  let $timeS = $container.find('.start-time input[data-datetime-part=time]');
+  let $timeE = $container.find('.end-time input[data-datetime-part=time]');
+  let isValid = true;
+  let errorMessage;
+
+  if (!($timeS.val() === '') === ($timeE.val() === '')) {
+    isValid = false;
+    errorMessage = 'Needs to set both or none';
+  } else if ($timeS.val() > $timeE.val()) {
+    isValid = false;
+    errorMessage = 'Needs to set both or none';
+    errorMessage = 'End time is before start time';
+  }
+  if (isValid) {
+    return true;
+  }
+  // will be refactored
+  $container.addClass('has-error');
+  $container.append(`<span class="help-block">${errorMessage}</span>`);
+  return false;
+};
```

Comment:
I think to solve problem this we need to force people to have some value in field.
If you want to clear, you must use button. But if you start edit it will automatically add all required values.

Reply:
WIll be added with delete button?